code/game/atoms_movable.dm

```diff
@@ -1650,3 +1652,19 @@
 */
 /atom/movable/proc/keybind_face_direction(direction)
 	setDir(direction)
+
+/**
+ * Check if the other atom/movable has any factions the same as us. Defined at the atom/movable level so it can be defined for just about anything.
+ *
+ * If exact match is set, then all our factions must match exactly
+ */
+/atom/movable/proc/faction_check_atom(atom/movable/target, exact_match)
+	if(exact_match) //if we need an exact match, we need to do some bullfuckery.
+		var/list/faction_src = faction.Copy()
+		var/list/faction_target = target.faction.Copy()
+		if(!("[REF(src)]" in faction_target)) //if they don't have our ref faction, remove it from our factions list.
+			faction_src -= "[REF(src)]" //if we don't do this, we'll never have an exact match.
+		if(!("[REF(target)]" in faction_src))
+			faction_target -= "[REF(target)]" //same thing here.
+		return faction_check(faction_src, faction_target, TRUE)
+	return faction_check(faction, target.faction, FALSE)
```

Comment:
this can be an early return instead

```suggestion
	if(exact_match)
		return faction_check(faction, target.faction, FALSE)

	var/list/faction_src = faction.Copy()
	var/list/faction_target = target.faction.Copy()
	if(!("[REF(src)]" in faction_target)) //if they don't have our ref faction, remove it from our factions list.
		faction_src -= "[REF(src)]" //if we don't do this, we'll never have an exact match.
	if(!("[REF(target)]" in faction_src))
		faction_target -= "[REF(target)]" //same thing here.
	return faction_check(faction_src, faction_target, TRUE)
```

Reply:
also faction appears to be a lazylist at several levels now, is this still safe for that?